frontend/src/metabase/visualizations/lib/scalar_utils.ts

```diff
@@ -15,21 +22,19 @@ export function compactifyValue(
     ...formatOptions,
     compact: false,
   });
-  const compactScalarValue = formatValue(value, {
-    ...formatOptions,
-    compact: true,
-  });
 
-  // use the compact version of formatting if the component is narrower than
-  // the cutoff and the formatted value is longer than the cutoff
-  // also if the width is less than a certain multiplier of the number of digits
-  const displayCompact =
-    fullScalarValue !== null &&
-    typeof fullScalarValue === "string" &&
-    fullScalarValue.length > COMPACT_MIN_LENGTH &&
-    (width < COMPACT_MAX_WIDTH ||
-      width < COMPACT_WIDTH_PER_DIGIT * fullScalarValue.length);
-  const displayValue = displayCompact ? compactScalarValue : fullScalarValue;
+  const canCompact = typeof fullScalarValue === "string";
+  if (!canCompact) {
+    return { displayValue: fullScalarValue, fullScalarValue };
+  }
+
+  const displayValue =
+    formatOptions.compact || checkShouldCompact(fullScalarValue, width)
+      ? formatValue(value, {
+          ...formatOptions,
+          compact: true,
+        })
+      : fullScalarValue;
 
   return { displayValue, fullScalarValue };
 }
```

Comment:
See the reply to [this comment](https://github.com/metabase/metabase/pull/36992/files#r1434440644) for the suggested changes here as well. It has to do with moving the compact visualization setting to `computeTrend` and outside of `SmartScalar`